framework/CodeInterfaces/RAVEN/RAVENInterface.py

```diff
@@ -0,0 +1,287 @@
+        except TypeError:
+          raise IOError(self.printTag+' ERROR: convertNotScalarSampledVariables accept only one argument convertNotScalarSampledVariables(variableDict)')
+      else:
+        print(self.printTag+' Warning: method "convertNotScalarSampledVariables" has been inputted but no "no scalar" variables have been found!')
+    # check if ext module has the method to manipulate the variables
+    if self.hasMethods['scalar']:
+      try:
+        self.extModForVarsManipulation.manipulateScalarSampledVariables(modifDict)
+      except TypeError:
+        raise IOError(self.printTag+' ERROR: manipulateScalarSampledVariables accept only one argument manipulateScalarSampledVariables(variableDict)')
+
+    # we work on batchSizes here
+    newBatchSize = Kwargs['NumMPI']
+    internalParallel = Kwargs.get('internalParallel',False)
+    if int(Kwargs['numberNodes']) > 0:
+      # we are in a distributed memory machine => we allocate a node file
+      nodeFileToUse = os.path.join(Kwargs['BASE_WORKING_DIR'],"node_" +str(Kwargs['INDEX']))
+      if os.path.exists(nodeFileToUse):
+        modifDict['RunInfo|mode'           ] = 'mpi'
+        modifDict['RunInfo|mode|nodefile'  ] = nodeFileToUse
+      else:
+        raise IOError(self.printTag+' ERROR: The nodefile "'+str(nodeFileToUse)+'" does not exist!')
+    if internalParallel or newBatchSize > 1:
+      # either we have an internal parallel or NumMPI > 1
+      modifDict['RunInfo|batchSize'       ] = newBatchSize
+    #modifDict['RunInfo|internalParallel'] = internalParallel
+    #make tree
+    modifiedRoot = parser.modifyOrAdd(modifDict,save=True,allowAdd = True)
+    #make input
+    parser.printInput(modifiedRoot,currentInputFiles[index].getAbsFile())
+    # copy slave files
+    parser.copySlaveFiles(currentInputFiles[index].getPath())
+    return currentInputFiles
+
+  def checkForOutputFailure(self,output,workingDir):
+    """
+      This method is called by the RAVEN code at the end of each run  if the return code is == 0.
+      This method needs to be implemented by the codes that, if the run fails, return a return code that is 0
+      This can happen in those codes that record the failure of the job (e.g. not converged, etc.) as normal termination (returncode == 0)
+      This method can be used, for example, to parse the outputfile looking for a special keyword that testifies that a particular job got failed
+      (e.g. in RAVEN would be the keyword "raise")
+      @ In, output, string, the Output name root
+      @ In, workingDir, string, current working dir
+      @ Out, failure, bool, True if the job is failed, False otherwise
+    """
+    failure = True
+    try:
+      outputToRead = open(os.path.join(workingDir,output),"r")
+    except:
+      return failure
+    readLines = outputToRead.readlines()
+    for badMessage in ["Traceback ","raise","IOError"]:
+      if not any(badMessage in x for x in readLines):
+        failure = False
```

Comment:
If "IOError" is not in the lines, this will have failure = False (even if say Traceback or raise is found).  I think you want:
```
foundBadMessage = False
 for badMessage in ["Traceback ","raise","IOError"]:
      if any(badMessage in x for x in readLines):
        foundBadMessage = True
if not foundBadMessage:
  failure = False
```